src/main/java/org/broadinstitute/consent/http/resources/ElectionResource.java

```diff
@@ -1,26 +1,73 @@
 package org.broadinstitute.consent.http.resources;
 
+import org.broadinstitute.consent.http.enumeration.VoteType;
 import org.broadinstitute.consent.http.models.Election;
+import org.broadinstitute.consent.http.models.Vote;
 import org.broadinstitute.consent.http.models.dto.Error;
 import org.broadinstitute.consent.http.service.AbstractElectionAPI;
+import org.broadinstitute.consent.http.service.AbstractVoteAPI;
 import org.broadinstitute.consent.http.service.ElectionAPI;
+import org.broadinstitute.consent.http.service.VoteAPI;
 
 import javax.annotation.security.PermitAll;
 import javax.annotation.security.RolesAllowed;
-import javax.ws.rs.*;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
 import javax.ws.rs.core.Context;
 import javax.ws.rs.core.Response;
 import javax.ws.rs.core.UriInfo;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
 
 @Path("{api : (api/)?}election/")
 public class ElectionResource extends Resource {
 
     private final ElectionAPI api;
+    private final VoteAPI voteApi;
 
     public ElectionResource() {
         this.api = AbstractElectionAPI.getInstance();
+        this.voteApi = AbstractVoteAPI.getInstance();
     }
 
+    @POST
+    @Consumes("application/json")
+    @Produces("application/json")
+    @Path("/{referenceId}/advance/{vote}")
+    @RolesAllowed({ADMIN})
+    public Response advanceElection(@PathParam("referenceId") String referenceId, @PathParam("vote") String vote) {
+        try {
+            List<Vote> voteList = voteApi.describeVotes(referenceId);
+            List<Integer> electionIds = voteList.stream().map(Vote::getElectionId).collect(Collectors.toList());
+            Date now = new Date();
+            Boolean voteValue = vote.equalsIgnoreCase("yes");
+            voteList.
```

Comment:
Just a thought. Maybe this logic can be moved to a service

Reply:
Yes, the should, but I hesitate to add to current services as they are constructed now - the election and vote api classes use the old pattern and I don't want to tackle refactoring those in this PR. I'm on the fence, but lean towards leaving them as is here and tackling the refactor later. The API classes are so convoluted and inefficient, they really need a complete re-write.

Reply:
Ok, you convinced me @vvicario - I'm going to take a shot at extracting out a new service class and add some TODOs for migrating older code to newer.